### PR TITLE
Refactor salesforce driver

### PR DESCRIPTION
**Pull request by maretskiy**

### CODE REVIEW

File: notify/drivers/sfdc.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+    def authenticate_soap(self):
+        LOG.debug("Making SFDC SOAP auth for {}".format(self.username))
+        doc = minidom.Document()
+        env = doc.appendChild(doc.createElement("soapenv:Envelope"))
+        env.setAttribute("xmlns:soapenv",
+                         "http://schemas.xmlsoap.org/soap/envelope/")
+        env.setAttribute("xmlns:urn", "urn:partner.soap.sforce.com")
+        header = env.appendChild(doc.createElement("soapenv:Header"))
+        a = header.appendChild(doc.createElement("urn:CallOptions"))
```
Comment (boris-42) [inline notify/drivers/sfdc.py]: Remove please hardcoded make a loop iterate over the elemennts to add them properly.

File: notify/drivers/sfdc.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+            return self._request(method, url, headers=headers, repeat=False,
+                                 **kwargs)
+
+        return response.status_code, response_json
+
+
+class Driver(base.Driver):
+    """SalesForce notofication driver."""
+
+    CONFIG_SCHEMA = {
```
Comment (boris-42) [inline notify/drivers/sfdc.py]: yep this is good stuff

**Comment by boris-42** [comment]
@maretskiy please avoid in future such messy patches. 

E.g. you should have here 2 PR: one that renames notify -> alerts another that refactors SF plugin

File: notify/driver.py [hunk 3]
```diff
@@ -85,14 +82,14 @@ class Driver(object):
     }
 
     @classmethod
-    def validate_payload(cls, payload):
```
Comment (boris-42) [inline notify/driver.py]: @maretskiy payload is way way way better term than alert https://en.wikipedia.org/wiki/Payload_(computing) and actually i would like to use it so please revert this change. 

Alert is not clear actually in this context

File: notify/drivers/sfdc.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+import requests
+from requests.packages.urllib3 import exceptions as urllib_exc
+
+from notify import config
+from notify import driver
+
+requests.packages.urllib3.disable_warnings(urllib_exc.InsecureRequestWarning)
+
+
+LOG = logging.getLogger("sfdc")
```
Comment (akscram) [inline notify/drivers/sfdc.py]: You should not name modules like that because it breaks hierarchical structure of logging and makes it painful to configure it later. So, I would suggest to use `__name__` here and this module will use the `notify.drivers.sfdc` logger. This convention will provide an the easiest way to configure logger, e.g. to set level, formatters or streams  for all drivers `notify.drivers.*` or just for this specific driver `notify.drivers.sfdc` in the future.

File: notify/drivers/sfdc.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+LOG.setLevel(config.get_config().get("logging", {}).get("level", "INFO"))
```
Comment (akscram) [inline notify/drivers/sfdc.py]: It should not be configured in this way right in the code, it have to be done through [the logging configuration](https://docs.python.org/3/library/logging.config.html), but if you want to set default logging level you have to specify:

```
LOG.setLevel(logging.INFO)
````
and nothing more because the whole logging should be configured in one place.
